spells/fireBall.lua

```diff
@@ -0,0 +1,56 @@
+FireballSpell = Spell:extend()
+
+function FireballSpell:new(character)
+	local cooldown = 4
+	self.super:new(cooldown)
+	self.character = character
+end
+
+function FireballSpell:use()
+	if self.curCooldown < 0 then
+		local entity = FireballEntity(self.character)
+
+		sceneManager.curScene:addEntity(entity)
+
+		self.curCooldown = self.cooldown		
+	end	
+end
+
+FireballEntity = SpellEntity:extend()
+
+function FireballEntity:new(character)
+	local activeTime = 3
+	self.super:new(activeTime)
+
+	self.x = character.x
+	self.y = character.y
+	self.dir = character.rot
+	self.speed = 200
+
+	self.shader = love.graphics.newShader[[
+	extern number time;
```

Comment:
Perhaps put shaders in a new file and directory, maybe something like `shaders/fireball`